app/views/handlers/confirmation_email.py

```diff
@@ -31,18 +51,49 @@ def get_page_title(self):
             return gettext("Error: {page_title}").format(page_title=self.page_title)
         return self.page_title
 
-    @staticmethod
-    def handle_post():
-        session_store = get_session_store()
-        session_store.session_data.confirmation_email_count += 1
-        session_store.save()
+    def _publish_fulfilment_request(self):
+        topic_id = current_app.config["EQ_SUBMISSION_CONFIRMATION_TOPIC_ID"]
+        fulfilment_request = ConfirmationEmailFulfilmentRequest(
+            self.form.email.data, self._session_data, self._schema
+        )
+        try:
+            return current_app.eq["publisher"].publish(
+                topic_id, message=fulfilment_request.payload
+            )
+        except PublicationFailed:
+            raise ConfirmationEmailFulfilmentRequestFailed
 
-    @staticmethod
-    def is_limit_reached():
-        session_store = get_session_store()
+    def handle_post(self):
+        self._publish_fulfilment_request()
+        self._session_store.session_data.confirmation_email_count += 1
+        self._session_store.save()
 
+    def _is_limit_reached(self):
         return (
-            session_store
-            and session_store.session_data.confirmation_email_count
+            self._session_data.confirmation_email_count
             >= current_app.config["CONFIRMATION_EMAIL_LIMIT"]
         )
+
+
+@dataclass
+class ConfirmationEmailFulfilmentRequest:
+    email_address: str
+    session_data: SessionData
+    schema: QuestionnaireSchema
+
+    @property
+    def payload(self):
+        message = {
+            "email_address": self.email_address,
+            "form_type": self.schema.form_type,
+            "region_code": self.schema.region_code,
+            "questionnaire_id": self.session_data.questionnaire_id,
+            "tx_id": self.session_data.tx_id,
+            "language_code": self.session_data.language_code,
+            "display_address": self.session_data.display_address,
+            "submitted_at": self.session_data.submitted_time,
+            "datetime": datetime.now(tzutc()).isoformat(),
+            "id": str(uuid4()),
+        }
```

Comment:
I'm confused by the `fulfilment` naming. This payload doesn't look anything like the `fulfilmentRequest` payload for the individual response. I think we might be better removing the use of `fulfilment` to differentiate it from fulfilment requests.

Reply:
As discussed, kept the word "fulfilment" in both to help it differentiate from runner http requests. They are now both treated a type of fulfilment request for which I introduced a basic ABC. The error handlers have now also been rationalised to be generic to fulfilment request while keeping track of what invoked it. Take a look at the latest changes and see what you think.